Packages/com.unity.inputsystem/InputSystem/Devices/ScreenKeyboard.cs

```diff
@@ -0,0 +1,155 @@
+using System;
+using System.Runtime.InteropServices;
+using UnityEngine.InputSystem.Utilities;
+
+
+namespace UnityEngine.InputSystem
+{
+    public enum ScreenKeyboardType
+    {
+        Default = 0,
+        ASCIICapable = 1,
+        NumbersAndPunctuation = 2,
+        URL = 3,
+        NumberPad = 4,
+        PhonePad = 5,
+        NamePhonePad = 6,
+        EmailAddress = 7,
+        Social = 8,
+        Search = 9
+    }
+
+    public enum ScreenKeyboardStatus
+    {
+        Visible,
+        Done,
+        Canceled,
+        LostFocus
+    }
+
+
+    [StructLayout(LayoutKind.Sequential)]
+    public struct ScreenKeyboardShowParams
+    {
+        public ScreenKeyboardType type;
+        public string initialText;
+        public string placeholderText;
+        public bool autocorrection;
+        public bool multiline;
+        public bool secure;
+
+        ////TODO: this one is iPhone specific?
+        public bool alert;
+
+        /// <summary>
+        /// Show keyboard without input field?
+        /// Only supported on iOS and Android.
+        /// Note: TODO, review this If input field is hidden, you won't receive inputFieldTextChanged callback, instead you'll be receiving onTextInput
+        /// </summary>
+        public bool inputFieldHidden;
+        ////TODO: no characterLimit here, because the logic for characterLimit is too complex when IME composition occurs, instead let user manage the text from OnTextChanged callbac
+    }
+
+
+    public class InputFieldEventArgs
+    {
+        public string text { set; get; }
+    }
+    ////TODO: probably need a better name, so not to collide with com.unity.inputsystem\InputSystem\Plugins\OnScreen\OnScreenKeyboard.cs
+    public class ScreenKeyboard : Keyboard
```

Comment:
Hmm, wondering... what's the usage we get out of basing this on `Keyboard`? Is any of the `Keyboard` functionality actually functional on `ScreenKeyboard`? If not, doesn't seem like it should be a device. Input devices are all based on a state and event model that doesn't really seem to apply to `ScreenKeyboard`.

Reply:
That's a good question,  need to think about it.

But there are some shared pieces:
- Keyboard.onTextInput, it will be called when ScreenKeyboard doesn't have an input pane
- Most of mobile keyboard don't support key events even though you click "keys"  on screen keyboard (you only receive text event), but I think there might be instances of screen keyboard which will support key events

Maybe I should try to rewrite the implementation with states, then it will be closer to other input devices.